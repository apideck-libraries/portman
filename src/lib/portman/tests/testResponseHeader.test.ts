import { OasMappedOperation, PostmanMappedOperation } from '../../'
import { getOasMappedOperation } from '../../../../__tests__/getOasMappedOperation'
import { getPostmanMappedOperation } from '../../../../__tests__/getPostmanMappedOperation'
import { testResponseHeader } from './testResponseHeader'

describe('testResponseHeader', () => {
  let oasOperation: OasMappedOperation
  let pmOperation: PostmanMappedOperation

  beforeEach(async () => {
    oasOperation = await getOasMappedOperation()
    pmOperation = await getPostmanMappedOperation()
  })

  it('should add test for response header', async () => {
    pmOperation = testResponseHeader('x-unify-request-id', pmOperation, oasOperation)
    const pmTest = pmOperation.getTests()
    expect(pmTest.script.exec).toMatchSnapshot()
  })
})
