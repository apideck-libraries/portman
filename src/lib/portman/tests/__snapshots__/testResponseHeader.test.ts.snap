// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testResponseHeader should add test for response header 1`] = `
Array [
  "// Validate if response header is present 
pm.test(\\"[GET]::/crm/companies/:id - Response header x-unify-request-id is present\\", function () {
   pm.response.to.have.header(\\"x-unify-request-id\\");
});
",
]
`;
